pattern/pattern.c

```diff
@@ -407,13 +407,16 @@ int mutt_pattern_func(struct MailboxView *mv, int op, char *prompt)
       /* new limit pattern implicitly uncollapses all threads */
       e->vnum = -1;
       e->visible = false;
+      e->limit_visited = false;
```

Comment:
Don't we want to set this to `true` no matter whether the pattern matches this email?